core/domain/exp_jobs_one_off_test.py

```diff
@@ -737,3 +737,79 @@ def test_migration_job_skips_deleted_explorations(self):
         # Ensure the exploration is still deleted.
         with self.assertRaisesRegexp(Exception, 'Entity .* not found'):
             exp_services.get_exploration_by_id(self.NEW_EXP_ID)
+
+class FallbackOneOffJobTest(test_utils.GenericTestBase):
+
+    EXP_IDS = ['exp_id0', 'exp_id1']
+    FALLBACK = [{
+        'trigger': {
+            'customization_args': {
+                'num_submits': {
+                    'value': i+3
+                }
+            },
+            'trigger_type': 'NthResubmission'
+        },
+        'outcome': {
+            'feedback': [
+                '<p>feedback %d</p>' % i
+            ],
+            'param_changes': [],
+            'dest': 'Introduction'
+        }
+    } for i in xrange(3)]
+    STATE_NAMES = ['Introduction', 'state 1']
+
+
+    def setUp(self):
+        super(FallbackOneOffJobTest, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+
+    def test_fallbacks_are_listed(self):
+        """Tests that fallbacks are listed.
+        """
+        explorations = [self.save_new_valid_exploration(
+            EXP_ID, self.owner_id, objective='The objective',
+            end_state_name='end') for EXP_ID in self.EXP_IDS]
+        for exploration in explorations:
+            init_state = exploration.states[exploration.init_state_name]
+            init_interaction = init_state.interaction
+            init_interaction.default_outcome.dest = 'state 1'
+            init_state.update_interaction_fallbacks(self.FALLBACK)
+            exploration.add_states(['state 1'])
+            new_state = exploration.states['state 1']
+            new_state.interaction.default_outcome.dest = 'end'
+            new_state.update_interaction_fallbacks(self.FALLBACK)
+            exploration.states['end'].update_interaction_id('EndExploration')
+
+            exp_services._save_exploration(self.owner_id, exploration, '', []) # pylint: disable=W0212
+            rights_manager.publish_exploration(self.owner_id, exploration.id)
+
+        job_id = (exp_jobs_one_off.FallbackOneOffJob.create_new())
+        exp_jobs_one_off.FallbackOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        actual_values = exp_jobs_one_off.FallbackOneOffJob.get_output(job_id)
+        updated_exps = [
+            exp_services.get_exploration_by_id(exploration.id)
+            for exploration in explorations]
+        expected_values = []
+
+        for updated_exp in updated_exps:
+            for state_name in self.STATE_NAMES:
+                interaction = updated_exp.states[state_name].interaction
+                expected_values.append([
```

Comment:
Actually here it's probably better to just hardcode the expected value directly. Otherwise you're just mimicking your implementation; you should instead be testing "if the function is given this input it produces this exact output".